py-polars/polars/utils/_construction.py

```diff
@@ -1441,8 +1441,32 @@ def arrow_to_pydf(
 ) -> PyDataFrame:
     """Construct a PyDataFrame from an Arrow Table."""
     original_schema = schema
+
+    if schema is None:
+        # Column names need to be passed as a dictionary to preserve empty
+        # string as a name. The names are also checked for duplicates during
+        # construction.
+        unpack_schema_input: SchemaDefinition = {}
+
+        for name in data.column_names:
+            if name in unpack_schema_input:
+                raise ValueError(
+                    f"Arrow table contained duplicate column '{name}'. Please "
+                    "rename or remove duplicate columns, or pass a schema to "
+                    "be used to override the column names. Note that the "
+                    "schema must be in the form of a dictionary if a column "
+                    "uses an empty string '' as its name. To automatically "
+                    "generate column names, pass a list of empty strings as "
+                    "the schema."
+                )
+            unpack_schema_input[name] = None  # type: ignore[index]
```

Comment:
Not sure how to resolve mypy error here - it complains with `Unsupported target for indexed assignment`, but `SchemaDefinition` type definition contains `type[None] | None`.